envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/LuaFilterFactoryTest.kt

```diff
@@ -37,4 +38,42 @@ internal class LuaFilterFactoryTest {
         assertThat(givenServiceName).isEqualTo(expectedServiceName)
         assertThat(givenDiscoveryServiceName).isEqualTo(expectedDiscoveryServiceName)
     }
+
+    @Test
+    fun `should create metadata with given flags`() {
+        // given
+        val flags = mapOf("flag1" to true, "flag2" to false)
+        val expectedServiceName = "service-1"
+        val expectedDiscoveryServiceName = "consul-service-1"
+        val group: Group = ServicesGroup(
+            communicationMode = CommunicationMode.XDS,
+            serviceName = expectedServiceName,
+            discoveryServiceName = expectedDiscoveryServiceName
+        )
+        val factory = LuaFilterFactory(IncomingPermissionsProperties())
+
+        // when
+        val metadata = factory.ingressScriptsMetadata(group, flags)
+        val luaMetadata = metadata.getFilterMetadataOrThrow("envoy.filters.http.lua")
+        // then
+        flags.forEach { (k, v) ->
+            assertThat(luaMetadata.getFieldsOrThrow("flags").structValue.getFieldsOrThrow(k).boolValue).isEqualTo(v)
+        }
+    }
+
+    @Test
+    fun `should leave flags empty`() {
+        val expectedServiceName = "service-1"
+        val expectedDiscoveryServiceName = "consul-service-1"
+        val group: Group = ServicesGroup(
+            communicationMode = CommunicationMode.XDS,
+            serviceName = expectedServiceName,
+            discoveryServiceName = expectedDiscoveryServiceName
+        )
+        val factory = LuaFilterFactory(IncomingPermissionsProperties())
+        // when
+        val metadata = factory.ingressScriptsMetadata(group)
```

Comment:
i believe it belongs to given part as when part shall have only desired interaction